lib/galaxy/webapps/galaxy/api/roles.py

```diff
@@ -44,10 +44,7 @@ class RoleListModel(BaseModel):
 def role_to_model(trans, role):
     item = role.to_dict(view='element', value_mapper={'id': trans.security.encode_id})
     role_id = trans.security.encode_id(role.id)
-    try:
-        item['url'] = url_for('role', id=role_id)
-    except AttributeError:
-        item['url'] = "*deprecated attribute not filled in by FastAPI server*"
+    item['url'] = url_for('role', id=role_id)
```

Comment:
What about using explicitly `handle_url_for()` here and in `lib/galaxy/managers/base.py`, while restoring `url_for = base.routes.url_for` in `lib/galaxy/web/framework/__init__.py` ?

Reply:
Thank you @nsoranzo,
I can do that, but my concern is more about what will happen when we totally move to FastAPI, and this `base.routes.url_for` will stop working for good. Will this affect all the tools using this kind of `baseurl` param?

Reply:
Sorry, I don't have an answer for that!

Reply:
Let's see if someone can shed some light ;)

Reply:
I think I meant for that comment above to be here, sorry.